### PR TITLE
Azure Cloud Provider Reconciliation

### LINKED ISSUE #8140: Implement Provider Reconciling for Azure

### PR DESCRIPTION
**Pull request by embik**
**What this PR does / why we need it**:

Improves the cloud provider for Azure to be capable of periodic reconciling, a feature introduced with #8101. It implements the extended provider functionality that enables periodic reconciling, and makes ownership decisions based on the `cluster` tag on resources that have tags. If the resource isn't owned by KKP, the resource isn't updated, and therefore enables customers to provide their custom cloud resources if they want. If a custom name is passed but the resource doesn't exist, the provider considers it a request to create and own that resource under a custom name.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #8140

**Special notes for your reviewer**:

This adds a few integration tests for resources that don't depend on `futures` during resource ensuring. I'm looking into how to mock Azure client interactions that include `futures` to wait for resource creation completion, but so far I haven't figured it out and it might not be possible.

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
Azure cloud resources are periodically reconciled
```

### CODE REVIEW

File: pkg/provider/cloud/azure/availability_set.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+	//
+	// Attributes we check:
+	// - SKU name
+	// - fault domain count
+	// - update domain count
+	if (availabilitySet.Sku != nil && availabilitySet.Sku.Name != nil && *availabilitySet.Sku.Name == *target.Sku.Name) && availabilitySet.AvailabilitySetProperties != nil &&
+		(availabilitySet.AvailabilitySetProperties.PlatformFaultDomainCount != nil && *availabilitySet.AvailabilitySetProperties.PlatformFaultDomainCount == *target.AvailabilitySetProperties.PlatformFaultDomainCount) &&
+		(availabilitySet.AvailabilitySetProperties.PlatformUpdateDomainCount != nil && *availabilitySet.AvailabilitySetProperties.PlatformUpdateDomainCount == *target.AvailabilitySetProperties.PlatformUpdateDomainCount) {
+		return cluster, nil
+	}
```
Comment (embik) [inline pkg/provider/cloud/azure/availability_set.go:69]: @moadqassem do you have any thoughts on this? I'm not happy with it, but due to the fact that Azure heavily relies on pointers, I can't do `reflect.DeepEqual` on the structs, at least that's my understanding.

Comment (moadqassem) [reply pkg/provider/cloud/azure/availability_set.go:69]: I am not sure tbh, but I ve seen quite few panics in the past when it came to azure :(

**Review by moadqassem — APPROVED** [review]
/approve
/lgtm

**Pull request merged**